python/pyspark/sql/catalog.py

```diff
@@ -237,23 +237,28 @@ def registerFunction(self, name, f, returnType=StringType()):
         :param name: name of the UDF
         :param f: python function
         :param returnType: a :class:`pyspark.sql.types.DataType` object
+        :return: a wrapped :class:`UserDefinedFunction`
 
-        >>> spark.catalog.registerFunction("stringLengthString", lambda x: len(x))
+        >>> strlen = spark.catalog.registerFunction("stringLengthString", len)
         >>> spark.sql("SELECT stringLengthString('test')").collect()
         [Row(stringLengthString(test)=u'4')]
 
+        >>> spark.sql("SELECT 'foo' AS text").select(strlen("text")).collect()
+        [Row(stringLengthString(text)=u'3')]
+
         >>> from pyspark.sql.types import IntegerType
-        >>> spark.catalog.registerFunction("stringLengthInt", lambda x: len(x), IntegerType())
+        >>> strlen = spark.catalog.registerFunction("stringLengthInt", len, IntegerType())
```

Comment:
Hm, I think capturing is rather better because it is what this API returns (and as it dose not return `None` anymore) if we don't have a better idea. Variable that indicates it's private, `_`, looks also fine to me.  No strong opinion. I am fine if we don't have a simpler and cleaner idea.